src/plugins/mer/merbuildconfiguration.cpp

```diff
@@ -0,0 +1,129 @@
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 as published by the Free Software
+** Foundation and appearing in the file LICENSE.LGPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU Lesser General Public License version 2.1 requirements
+** will be met: http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** Other Usage
+**
+** Alternatively, this file may be used in accordance with the terms and
+** conditions contained in a signed written agreement between you and Digia.
+**
+****************************************************************************/
+
+#include "merbuildconfiguration.h"
+
+#include "merconnection.h"
+#include "merconstants.h"
+#include "merbuildsteps.h"
+
+#include <projectexplorer/buildmanager.h>
+#include <projectexplorer/buildsteplist.h>
+#include <projectexplorer/projectexplorer.h>
+#include <qmakeprojectmanager/qmakeproject.h>
+#include <qmakeprojectmanager/qmakestep.h>
+
+#include <QTextStream>
+#include <QRegularExpression>
+
+using namespace ProjectExplorer;
+using namespace QmakeProjectManager;
+
+namespace {
+
+const char MER_VARIABLES_CACHE_FILENAME[] = ".mb2/qmake_variables.cache";
+
+QString parseVariableString(const QString &line)
+{
+    QRegExp rx("\\s*=\\s*");
+    const int index = line.indexOf(rx);
+    if (index == -1)
+        return QString();
+
+    const QString key = line.mid(0, index);
+    const QString value = line.mid(index + rx.matchedLength());
+
+    return QString("%1=%2").arg(key).arg(value);
+}
+
+QStringList readVariables(const QString &fileName)
+{
+    QStringList result;
+    QFile file(fileName);
+    if (file.exists() && file.open(QIODevice::ReadOnly | QIODevice::Text)) {
+        QTextStream in(&file);
+        while (!in.atEnd()) {
+            const QString line = parseVariableString(in.readLine());
+            if (line.size())
+                result << line;
+        }
+    }
+    return result;
+}
+
+} // namespace anonymous
+
+namespace Mer {
+namespace Internal {
+
+MerBuildConfiguration::MerBuildConfiguration(Target *target, Core::Id id)
+    : QmakeBuildConfiguration(target, id)
+    , m_useExtraParserArguments(false)
+{
+    QmakeProject *qmakeProject = static_cast<QmakeProject *>(project());
+    connect(qmakeProject, &QmakeProject::buildDirectoryInitialized,
+            this, &MerBuildConfiguration::setupExtraParserArguments);
+}
+
+void MerBuildConfiguration::setUseExtraParserArguments(bool value)
+{
+    m_useExtraParserArguments = value;
+    setupExtraParserArguments();
+}
+
+void MerBuildConfiguration::updateExtraParserArguments() const
+{
+    BuildStepList *steps = stepList(Core::Id(ProjectExplorer::Constants::BUILDSTEPS_BUILD));
+    QTC_ASSERT(steps, return);
+    const auto sdkStartStep = qobject_cast<Mer::Internal::MerSdkStartStep *>(steps->at(0));
```

Comment:
> BTW, if we are adding this class, we should get rid of MerAddVmStartBuildStepProjectListener

I thinked about this (base class - MerProjectListener may be omitted too). Then I was confused by this comment in meraddvmstartbuildstepprojectlistener.h: [it is not possible to simply override QmakeBuildConfigurationFactory::restore()...](https://github.com/sailfishos/sailfish-qtcreator/blob/eff7ab4e11e0fcf7fdbf80827acf4b0bfaa63462/src/plugins/mer/meraddvmstartbuildstepprojectlistener.h#L35) At the first glance it may be outdated, but I decided to leave this class in place. Please, share your thoughts about it.

Reply:
IIRC the "in order to..." was mean to continue "...to upgrade existing project configurations (those created before SDK upgrade) adding the newly introduced 'start VM' build steps to them". This happened long time ago, so it should be safe to only care about initializing new configurations now. Yes, the base class may be omitted to - do that with a separate commit please (who knows when we need it again :))

Reply:
Done.